upbound_uxp/datadog_checks/upbound_uxp/check.py

```diff
@@ -0,0 +1,590 @@
+                    for sample in family.samples:
+                        name = sample[NAME_IDX]
+                        tags = self._labels_to_tags(sample[LABEL_IDX], pod.metadata.name)
+                        value = sample[VALUE_IDX]
+
+                        if self.metrics_default != 'max' and name not in self.metrics_set:
+                            continue
+
+                        if name in metrics_map_keys:
+                            if self.verbose:
+                                print("Sending metric: " + name + " as: " + self.metrics_map[name])
+                            name = self.metrics_map[name]
+                        name = metrics_prefix + name
+
+                        if name.endswith('sum'):
+                            metric_type = 'summary'
+                        if name.endswith('count'):
+                            metric_type = 'counter'
+                        if name.endswith('bucket'):
+                            metric_type = 'histogram'
+
+                        # counter: _total
+                        if metric_type == 'counter':
+                            if self.verbose:
+                                print("%s: Name: %s, Tags: %s, Value: %s" % (metric_type, name, tags, str(value)))
+                            try:
+                                self.count(name, value, tags=tags)
+                            except Exception as e:
+                                print(e)
+                        # histogram: _bucket
+                        elif metric_type == 'histogram':
+                            if self.verbose:
+                                print("%s: Name: %s, Tags: %s, Value: %s" % (metric_type, name, tags, str(value)))
+                            try:
+                                self.histogram(name, value, tags=tags)
+                            except Exception as e:
+                                print(e)
+                        # summary: _sum
+                        elif metric_type == 'summary':
+                            if self.verbose:
+                                print("%s: Name: %s, Labels: %s, Value: %s" % (metric_type, name, tags, str(value)))
+                            try:
+                                self.count(name, value, tags=tags)
+                            except Exception as e:
+                                print(e)
+                        # gauge: no _bucket, _sum, _total postfix
+                        elif metric_type == 'gauge':
+                            if self.verbose:
+                                print("%s: Name: %s, Tags: %s, Value: %s" % (metric_type, name, tags, str(value)))
+                            try:
+                                self.gauge(name, value, tags=tags)
+                            except Exception as e:
+                                print(e)
+                        else:
+                            print('WARNING: metric type ' + metric_type + ' unkown for ' + name)
```

Comment:
Since this integration seems to be obtaining and parsing metrics in Prometheus format, I think we should try to get it to use the implemented functionality already provided with the `OpenMetricsBaseCheckV2` base class. For reference, see:

- https://datadoghq.dev/integrations-core/base/openmetrics/
- https://docs.datadoghq.com/developers/custom_checks/prometheus/

Using it should simplify the implementation of this integration significantly and benefit from any improvements that our team might implement in there.

Reply:
@alopezz It would be great to receive guidance on how to fix the .github/CODEOWNERS file and how to obtain the datadog API key during the automated github tests to get the current version passing all the way first.

Once everything passes, we can consider to change the implementation if the open metrics check is compatible with the data feed.